specreduce/utils/utils.py

```diff
@@ -0,0 +1,181 @@
+import copy
+
+import numpy as np
+
+from specreduce.core import _ImageParser
+from specreduce.tracing import Trace, FlatTrace
+from specreduce.extract import _ap_weight_image
+
+
+def measure_cross_dispersion_profile(image, trace=None, crossdisp_axis=0,
+                                     width=None, pixel=None, pixel_range=None,
+                                     statistic='median'):
+    """
+    Return a 1D (quantity) array of the cross-dispersion profile measured at a
+    specified pixel value ('wavelength', but potentially before calibration),
+    or the average profile across several pixel values (or range of pixel values)
+    along the dispersion axis.
+
+    If a single number is specified for `pixel`, then the profile at that pixel
+    (i.e wavelength) will be returned. If several pixels are specified in a list or
+    array, then they will be averaged (median or mean, set by `statistic` which
+    defaults to median). Alternatively, `pixel_range` can be specified as a tuple
+    of integers specifying the minimum and maximum pixel range to average the
+    profile. `pixel` and `pixel_range` cannot be set simultaneously, and the default
+    is `pixel_range`=(min_image, max_image) to return an average profile across the
+    entire wavelength range of the image.
+
+    The window in the cross dispersion axis for measuring the profile at
+    the pixel(s) specified is determined by `width` and `trace`. If a trace is
+    provided (either as a Trace object, or as a number specifying the location
+    on the cross-dispersion axis of a FlatTrace object) that will determine the
+    center of the profile on the cross-dispersion axis. Otherwise, if `trace`
```

Comment:
is this done by shifting pixels or does it support subpixel via interpolation, etc?  Or - unless this logic is within some other function that I forget about - is this just using the trace to create the mask but then not doing any recentering/shifting at all?

Reply:
This is just using the trace to create the mask

Reply:
Is that sufficient?  I thought I remember hearing the request for subpixel support along the trace, but can't remember if that was deferred or not.

Reply:
To follow-up: the plan for now will be to implement an option for aligning by the trace by pixel-shifting (integer, not subpixel).

Reply:
i added this functionality with the 'align_along_trace' arg